File: first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int count[30] = {0};
        int indexes[30] = {0};
        for(int i=0;i<s.length();i++)
        {
            count[s[i]-'a']++;
            indexes[s[i]-'a'] = i;
        }
        
        int ans = s.length();
        for(int i=0;i<26;i++)
            if(count[i] == 1)
                ans = min(ans,indexes[i]);
        if(ans == s.length())return -1;
        return ans;
    }
};

static int _ = [](){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    return 0;
}();

File: May LeetCoding Challenge/First Unique Character in a String.java
class Solution {
    public int firstUniqChar(String s) {
        int[] count = new int[30];
        Arrays.fill(count,-1);
        char[] chr = s.toCharArray();
        for(int i=0;i<chr.length;i++)
        {
            if(count[chr[i]-'a'] == -1)
                count[chr[i]-'a'] = i;
            else
                count[chr[i]-'a'] = -2;
        }
        boolean flag = true;
        int mn = Integer.MAX_VALUE;
        for(int i=0;i<26;i++)
        {
            if(count[i] >= 0)
            {
                flag = false;
                mn = Math.min(mn,count[i]);
            }
        }
        if(flag)return -1;
        return mn;
    }
}
